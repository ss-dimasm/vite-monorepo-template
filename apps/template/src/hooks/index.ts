/* istanbul ignore file */
export * from './use-platform-get'
export * from './use-platform-update'
export * from './utils'
