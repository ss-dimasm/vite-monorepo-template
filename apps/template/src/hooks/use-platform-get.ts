import { useEffect, useMemo } from 'react'
import {
  StringMap,
  getMergedHeaders,
  handleReapitError,
  getUrl,
  RC_SESSION_MISSING_ERROR,
  NETWORK_ERROR,
} from './utils'
import { useSnack } from '@reapit/elements'
import { useQuery, useQueryClient } from '@tanstack/react-query'
import axios, { AxiosError } from 'axios'
import { NavigateFunction, useNavigate } from 'react-router'
import { reapitConnectBrowserSession } from '../core/connect-session'

export type PlatformGet<DataType> = [
  data: DataType | null,
  loading: boolean,
  error: string | null,
  refresh: (queryParams?: Object) => void,
  refreshing: boolean,
  clearCache: () => void,
]

export interface UsePlatformGetParams {
  path: string
  errorMessage?: string
  successMessage?: string
  queryParams?: Object
  headers?: StringMap
  fetchWhenTrue?: any[]
  retry?: number | boolean
  onSuccess?: (message: string) => void
  onError?: (message: string) => void
}

export const handleError =
  (
    isError: boolean,
    error: AxiosError<any, any> | null,
    errorSnack: (text: string, timeout?: number | undefined) => void,
    navigate: NavigateFunction,
    errorMessage?: string,
    onError?: (message: string) => void,
  ) =>
  () => {
    if (isError && error) {
      const isRcError = error.message === RC_SESSION_MISSING_ERROR
      const isFourOOne = error.code === NETWORK_ERROR
      if (isRcError || isFourOOne) {
        return navigate('/login')
      }

      const errorString = handleReapitError(error, errorMessage)
      if (onError && errorString) onError(errorString)
      if (!onError && errorString) errorSnack(errorString)
      console.error(errorString)
    }
  }

export const handleSuccess =
  (
    isSuccess: boolean,
    successSnack: (text: string, timeout?: number | undefined) => void,
    successMessage?: string,
    onSuccess?: (message: string) => void,
  ) =>
  () => {
    if (isSuccess) {
      if (onSuccess && successMessage) onSuccess(successMessage)
      if (!onSuccess && successMessage) successSnack(successMessage)
    }
  }

export const usePlatformGet = <DataType>({
  path,
  errorMessage,
  successMessage,
  queryParams,
  headers,
  fetchWhenTrue,
  onSuccess,
  onError,
  retry,
}: UsePlatformGetParams): PlatformGet<DataType> => {
  const { success: successSnack, error: errorSnack } = useSnack()
  const navigate = useNavigate()
  const queryClient = useQueryClient()

  const isEnabled = fetchWhenTrue?.length
    ? Boolean(fetchWhenTrue?.filter(Boolean).length === fetchWhenTrue.length)
    : true

  const url = useMemo(getUrl(path, queryParams), [path, queryParams])

  const { data, error, isLoading, isSuccess, isError, isRefetching, refetch } = useQuery<DataType, AxiosError<any>>({
    queryKey: [url],
    queryFn: async () => {
      const reqHeaders = await getMergedHeaders(reapitConnectBrowserSession, headers)

      if (!reqHeaders) throw new Error(RC_SESSION_MISSING_ERROR)

      const req = await axios.get<DataType>(url, {
        headers: reqHeaders,
      })
      return req.data
    },
    retry: retry === undefined ? 1 : retry,
    refetchOnWindowFocus: false,
    enabled: isEnabled,
  })

  useEffect(handleSuccess(isSuccess, successSnack, successMessage, onSuccess), [
    isSuccess,
    successMessage,
    onSuccess,
    successSnack,
  ])

  useEffect(handleError(isError, error, errorSnack, navigate, errorMessage, onError), [
    isError,
    error,
    errorMessage,
    onError,
    errorSnack,
    navigate,
  ])

  const result = data ? data : null
  const errorString = error?.message ? error.message : null
  const loading = isEnabled && isLoading
  const clearCache = () => {
    queryClient.removeQueries({ queryKey: [url] })
  }

  return [result, loading, errorString, refetch, isRefetching, clearCache]
}
