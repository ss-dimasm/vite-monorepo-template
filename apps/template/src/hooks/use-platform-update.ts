import { useMemo } from 'react'
import { useSnack } from '@reapit/elements'
import {
  getMergedHeaders,
  getUrl,
  handleReapitError,
  NETWORK_ERROR,
  RC_SESSION_MISSING_ERROR,
  StringMap,
} from './utils'
import { useMutation } from '@tanstack/react-query'
import axios, { AxiosError, AxiosResponseHeaders } from 'axios'
import { useNavigate } from 'react-router'
import { reapitConnectBrowserSession } from '../core/connect-session'

export type ReapitUpdateState<ParamsType, DataType> = [
  UpdateFunction<ParamsType, DataType>,
  boolean,
  DataType | null,
  boolean,
  boolean,
]

type AcceptedMethod = 'POST' | 'PUT' | 'PATCH' | 'DELETE' | 'GET'

export interface ReapitUpdate {
  path: string
  method?: AcceptedMethod
  shouldReturnRecord?: boolean
  headers?: StringMap
  errorMessage?: string
  successMessage?: string
}

export type UpdateFunction<ParamsType, DataType> = (params: ParamsType) => Promise<DataType>

export const usePlatformUpdate = <ParamsType, DataType>({
  path,
  method = 'POST',
  shouldReturnRecord = false,
  headers = {},
  errorMessage,
  successMessage,
}: ReapitUpdate): ReapitUpdateState<ParamsType, DataType> => {
  const { error: errorSnack, success: successSnack } = useSnack()
  const navigate = useNavigate()
  const url = useMemo(getUrl(path), [path])

  const { mutateAsync, data, isSuccess, isError, isPending } = useMutation<DataType, AxiosError<any>, ParamsType>({
    mutationKey: [url],
    mutationFn: async (data: ParamsType) => {
      const updateHeaders = await getMergedHeaders(reapitConnectBrowserSession, headers)

      if (!updateHeaders) throw new Error(RC_SESSION_MISSING_ERROR)

      const res = await axios<DataType>(url, {
        method,
        headers: updateHeaders,
        data,
      })

      if (!shouldReturnRecord) return true

      const resHeaders = res.headers as AxiosResponseHeaders
      const location = resHeaders.get('Location')?.valueOf() as string

      if (!location) throw new Error('Location was not returned by server')

      const locationUrl = location.includes('.prod.paas') ? location.replace('.prod.paas', '') : location

      const locationRes = await axios(locationUrl, {
        method: 'GET',
        headers: updateHeaders,
      })

      return locationRes.data
    },
    onSuccess: () => {
      if (successMessage) successSnack(successMessage)
    },
    onError: (error: AxiosError<any>) => {
      const isRcError = error?.message === RC_SESSION_MISSING_ERROR
      const isFourOOne = error.code === NETWORK_ERROR

      if (isRcError || isFourOOne) {
        return navigate('/login')
      }

      const errorString = !isRcError ? handleReapitError(error, errorMessage) : null
      if (errorString) errorSnack(errorString, 5000)
      console.error(errorString)
    },
  })

  const updateFunction: UpdateFunction<ParamsType, DataType> = (data: ParamsType) => mutateAsync(data)
  const returnData = data ?? null

  return [updateFunction, isPending, returnData, isSuccess, isError]
}
