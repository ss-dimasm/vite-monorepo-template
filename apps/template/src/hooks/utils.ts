import { ReapitConnectBrowserSession } from '@reapit/connect-session'
import { AxiosError } from 'axios'
import qs from 'qs'

export const RC_SESSION_MISSING_ERROR =
  'Missing valid Reapit Connect Session, please try logging in again if the problem persists'
export const NETWORK_ERROR = 'ERR_NETWORK'

export interface StringMap {
  [key: string]: string
}

export interface ReapitErrorField {
  field?: string
  message?: string
}

export interface ReapitError {
  statusCode?: number
  errors?: ReapitErrorField[]
  description?: string
  dateTime?: string
  message?: string
  title?: string
}

export const getMergedHeaders = async (
  reapitConnectBrowserSession: ReapitConnectBrowserSession,
  headers?: StringMap,
): Promise<StringMap | null> => {
  const connectSession = await reapitConnectBrowserSession.connectSession()
  const accessToken = connectSession?.accessToken

  return accessToken
    ? {
        Authorization: `Bearer ${accessToken}`,
        'api-version': 'latest',
        'Content-Type': 'application/json',
        ...headers,
      }
    : null
}

export const handleReapitError = (error: AxiosError<any>, defaultMessage?: string): string => {
  const reapitError: ReapitError = error?.response?.data
  const { description, title, message, errors } = reapitError ?? {}
  const messageString = description
    ? description
    : title
      ? title
      : message
        ? message
        : error?.message
          ? error.message
          : defaultMessage
            ? defaultMessage
            : 'An unknown error has occurred, please refresh the page and try again.'
  const fieldErrors = Array.isArray(errors) ? errors?.map(({ field, message }) => `"${field}: ${message}"`) : null
  const fieldString = fieldErrors ? fieldErrors.join(', ') : ''

  return `${messageString} ${fieldString}`
}

export const stringListToBatchQuery = (list: (string | number)[], queryKey: string): string =>
  list.reduce((query: string, nextItem: string | number, index: number) => {
    if (!index) {
      return String(nextItem)
    }
    return `${query}&${queryKey}=${nextItem}`
  }, '')

// Where you have an object of filters with a mixture of arrays and strings / boolean / numeric values and
// you need a string map returned for useReapitGet
export const objectToQuery = <QueryObjectType extends {}>(queryObject: QueryObjectType): StringMap =>
  Object.keys(queryObject).reduce((currentQuery: StringMap, nextItem: string) => {
    const objectItem = queryObject[nextItem]

    if (Array.isArray(objectItem)) {
      currentQuery[nextItem] = stringListToBatchQuery(objectItem, nextItem)
    } else if (objectItem) {
      currentQuery[nextItem] = String(objectItem)
    }

    return currentQuery
  }, {} as StringMap)

export const getUrl = (path: string, queryParams?: Object) => (): string => {
  const api = process.env.PLATFORM_API_URL
  const normalisedQuery = objectToQuery(queryParams ?? {})
  const query = qs.stringify(normalisedQuery, { encode: false })
  const url = `${api}${path}${query ? `?${query}` : ''}`
  return url
}
