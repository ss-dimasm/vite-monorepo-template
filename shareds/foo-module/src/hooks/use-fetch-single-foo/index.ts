interface UseFetchSingleFooOptions {
  embedBaz?: boolean
}

export const useFetchSingleFoo = (fooId: string, options: UseFetchSingleFooOptions) => {
  const { embedBaz } = options ?? {}

  return {
    data: fooId,
    isLoading: false,
    metadata: {
      bazEntity: embedBaz ? 'baz-entity' : null,
    },
  }
}
