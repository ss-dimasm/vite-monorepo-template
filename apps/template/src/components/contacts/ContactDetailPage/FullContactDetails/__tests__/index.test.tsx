import { Mock } from 'vitest'
import { FullContactDetails } from '..'
import { usePlatformGet } from '../../../../../hooks'
import {
  mockSourceModelPagedResult,
  mockContactCategories,
  mockNegotiatorModelPagedResult,
  mockOfficeModelPagedResult,
  mockContactModel,
} from '../../../../../scripts/stubs'
import { render } from '../../../../../scripts/tests'

vi.mock('../../../../../hooks', () => ({
  usePlatformGet: vi.fn(),
  usePlatformUpdate: vi.fn(() => [vi.fn()]),
}))

const mockUsePlatformGet = usePlatformGet as Mock

describe('FullContactDetails', () => {
  it('should match a snapshot with data', () => {
    vi.mocked(mockUsePlatformGet)
      .mockReturnValue([mockSourceModelPagedResult])
      .mockReturnValue([mockContactCategories])
      .mockReturnValue([mockNegotiatorModelPagedResult])
      .mockReturnValue([mockOfficeModelPagedResult])

    expect(render(<FullContactDetails contact={mockContactModel} />).asFragment()).toMatchSnapshot()
  })

  it('should match a snapshot with no data', () => {
    expect(render(<FullContactDetails contact={null} />).asFragment()).toMatchSnapshot()
  })
})
