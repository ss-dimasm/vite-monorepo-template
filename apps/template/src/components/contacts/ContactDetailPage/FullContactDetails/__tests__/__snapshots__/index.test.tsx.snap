// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FullContactDetails > should match a snapshot with data 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
  <div
    class="el-fade-in el-grid"
  >
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name contact"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 15c-4.42 0-8 2.24-8 5h16c0-2.76-3.58-5-8-5m0-11a3.49 3.49 0 0 0-3.5 3.5c.002.754.069 1.507.2 2.25A3.48 3.48 0 0 0 12 13a3.48 3.48 0 0 0 3.3-3.25c.131-.743.198-1.496.2-2.25A3.49 3.49 0 0 0 12 4"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Name
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            Mr Foo Bar
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name info"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 2C6.477 2 2 6.477 2 12s4.477 10 10 10 10-4.477 10-10A10 10 0 0 0 12 2m1.143 15.714h-2.286v-6h2.286zm-1.143-8a1.714 1.714 0 1 1 0-3.428 1.714 1.714 0 0 1 0 3.428"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Active Status
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            <span
              class="el-intent-success el-status-indicator"
            />
             Active 
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name email"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M20 4H4a2 2 0 0 0-2 2v12a2 2 0 0 0 2 2h16a2 2 0 0 0 2-2V6a2 2 0 0 0-2-2m-1.41 2L12 12.59 5.41 6zM4 18V7.41L10.59 14a2 2 0 0 0 2.82 0L20 7.41V18z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Email
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            foo@bar.com
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name phone"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 16.81a3.3 3.3 0 0 0-.3-.81l-6-1a3.5 3.5 0 0 0-.85 1.78 10.74 10.74 0 0 1-6.65-6.63A3.5 3.5 0 0 0 9 9.29l-1-6a3.2 3.2 0 0 0-.83-.24A3 3 0 0 0 6.6 3 3.6 3.6 0 0 0 3 6.6C3 14.553 9.447 21 17.4 21a3.6 3.6 0 0 0 3.6-3.6 4 4 0 0 0 0-.59"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Mobile
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            242423424
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name phone"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 16.81a3.3 3.3 0 0 0-.3-.81l-6-1a3.5 3.5 0 0 0-.85 1.78 10.74 10.74 0 0 1-6.65-6.63A3.5 3.5 0 0 0 9 9.29l-1-6a3.2 3.2 0 0 0-.83-.24A3 3 0 0 0 6.6 3 3.6 3.6 0 0 0 3 6.6C3 14.553 9.447 21 17.4 21a3.6 3.6 0 0 0 3.6-3.6 4 4 0 0 0 0-.59"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Home
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            234234234
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name phone"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 16.81a3.3 3.3 0 0 0-.3-.81l-6-1a3.5 3.5 0 0 0-.85 1.78 10.74 10.74 0 0 1-6.65-6.63A3.5 3.5 0 0 0 9 9.29l-1-6a3.2 3.2 0 0 0-.83-.24A3 3 0 0 0 6.6 3 3.6 3.6 0 0 0 3 6.6C3 14.553 9.447 21 17.4 21a3.6 3.6 0 0 0 3.6-3.6 4 4 0 0 0 0-.59"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Work
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            2342342344
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name property"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m22.923 9.165-10.5-7a1 1 0 0 0-1.1 0l-10.45 7a1 1 0 0 0 1.1 1.66l.9-.63v11.3a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-11.3l.95.63a.94.94 0 0 0 .55.17 1 1 0 0 0 .55-1.83m-4.05 11.33h-14V8.865l7-4.67 7 4.67z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Primary Address
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            Bar Foo Foo Bar Foo Bar Foo
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name property"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m22.923 9.165-10.5-7a1 1 0 0 0-1.1 0l-10.45 7a1 1 0 0 0 1.1 1.66l.9-.63v11.3a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-11.3l.95.63a.94.94 0 0 0 .55.17 1 1 0 0 0 .55-1.83m-4.05 11.33h-14V8.865l7-4.67 7 4.67z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Secondary Address
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            Bar Foo Foo Bar Foo Bar Foo
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name property"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m22.923 9.165-10.5-7a1 1 0 0 0-1.1 0l-10.45 7a1 1 0 0 0 1.1 1.66l.9-.63v11.3a1 1 0 0 0 1 1h16a1 1 0 0 0 1-1v-11.3l.95.63a.94.94 0 0 0 .55.17 1 1 0 0 0 .55-1.83m-4.05 11.33h-14V8.865l7-4.67 7 4.67z"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Work Address
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            Bar Foo Foo Bar Foo Bar Foo
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name contact"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 15c-4.42 0-8 2.24-8 5h16c0-2.76-3.58-5-8-5m0-11a3.49 3.49 0 0 0-3.5 3.5c.002.754.069 1.507.2 2.25A3.48 3.48 0 0 0 12 13a3.48 3.48 0 0 0 3.3-3.25c.131-.743.198-1.496.2-2.25A3.49 3.49 0 0 0 12 4"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Negotiators
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            <div
              class="mocked-styled-166 el-badge-group"
            >
              <div
                class="mocked-styled-167 el-badge-group-inner"
              >
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name contacts"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M18 14.5a2.49 2.49 0 0 0 2.35-2.35q.141-.819.15-1.65a2.5 2.5 0 0 0-5 0q.009.831.15 1.65A2.49 2.49 0 0 0 18 14.5M8 15c-4.42 0-8 2.24-8 5h16c0-2.76-3.58-5-8-5m0-2a3.48 3.48 0 0 0 3.3-3.25c.131-.743.198-1.496.2-2.25a3.5 3.5 0 1 0-7 0c.002.754.069 1.507.2 2.25A3.48 3.48 0 0 0 8 13m10 3a8.4 8.4 0 0 0-1.61.15A5.63 5.63 0 0 1 18 20h6c0-2.21-2.69-4-6-4"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Offices
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            <div
              class="mocked-styled-166 el-badge-group"
            >
              <div
                class="mocked-styled-167 el-badge-group-inner"
              >
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                >
                  London Estate Agent
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name task"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 11a1 1 0 1 0 0 2 1 1 0 0 0 0-2m0-4a1 1 0 1 0 0-2 1 1 0 0 0 0 2m0 10a1 1 0 1 0 0 2 1 1 0 0 0 0-2m16-6H8a1 1 0 1 0 0 2h12a1 1 0 1 0 0-2M8 7h12a1 1 0 1 0 0-2H8a1 1 0 0 0 0 2m12 10H8a1 1 0 1 0 0 2h12a1 1 0 1 0 0-2"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Categories
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            <div
              class="mocked-styled-166 el-badge-group"
            >
              <div
                class="mocked-styled-167 el-badge-group-inner"
              >
                Unknown
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name cloud"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M6.5 20q-2.275 0-3.887-1.575T1 14.575q0-1.95 1.175-3.475Q3.349 9.575 5.25 9.15q.625-2.3 2.5-3.725T12 4q2.925 0 4.962 2.037T19 11q1.725.2 2.863 1.488A4.4 4.4 0 0 1 23 15.5q0 1.875-1.312 3.188Q20.375 20 18.5 20zm0-2h12q1.05 0 1.775-.725T21 15.5t-.725-1.775T18.5 13H17v-2q0-2.075-1.463-3.537Q14.075 6 12 6 9.925 6 8.463 7.463 7 8.925 7 11h-.5q-1.45 0-2.475 1.025A3.37 3.37 0 0 0 3 14.5q0 1.45 1.025 2.475A3.37 3.37 0 0 0 6.5 18"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Sources
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            <div
              class="mocked-styled-166 el-badge-group"
            >
              <div
                class="mocked-styled-167 el-badge-group-inner"
              >
                Unknown
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="mocked-styled-31 el-col"
    >
      <div
        class="el-flex-container"
      >
        <div
          class="mocked-styled-173 el-key-value-icon-wrap"
        >
          <span
            class="el-intent-primary el-icon-new el-icon"
          >
            <svg
              fill="none"
              height="1em"
              role="img"
              title="Icon image with name message"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 3H8a7 7 0 0 0 0 14h.88l1.78 3.55a1.44 1.44 0 0 0 2.68 0L15.12 17H16a7 7 0 1 0 0-14m0 12h-2.12L12 18.76 10.12 15H8A5 5 0 0 1 8 5h8a5 5 0 0 1 0 10"
                fill="currentColor"
              />
            </svg>
          </span>
        </div>
        <div
          class="el-flex-container el-flex-column"
        >
          <div
            class="el-text-base el-text-xs el-has-grey-text"
          >
            Marketing Preferences
          </div>
          <div
            class="el-text-base el-text-sm"
          >
            <div
              class="mocked-styled-166 el-badge-group"
            >
              <div
                class="mocked-styled-167 el-badge-group-inner"
              >
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                >
                  Marketing Consent
                </span>
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                >
                  Prefers Letter
                </span>
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                >
                  Prefers Email
                </span>
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                >
                  Prefers Phone
                </span>
                <span
                  class="el-intent-primary el-badge"
                  role="status"
                >
                  Prefers SMS
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`FullContactDetails > should match a snapshot with no data 1`] = `
<DocumentFragment>
  <div
    class="mocked-styled-93 el-snack-holder"
  />
</DocumentFragment>
`;
